### ISSUE TITLE
Missing JAR files in GAE package for ClientResource support on Google App Engine

### ISSUE THREAD

**Issue by rcbeuker** [restlet/restlet-framework-java#1118]
I have a Restlet server application on Google App Engine  that needs to call other resources (via HTTPS). When i just try the ClientResource i get an error;

>"The protocol used by this request is not declared in the list of client connectors. (HTTPS/1.1). In case you are using an instance of the Component class, check its "clients" property."

On stackoverflow there's a post about this; 

http://stackoverflow.com/questions/14206607/calling-restlet-client-resource-from-server-resource-causes-http-404-error

I'm running Restlet from a Servlet with a ServletAdapter so i've access to the component;

```java
ServletContext lServletContext = getServletContext();
ServletAdapter lServletAdapter = new ServletAdapter(lServletContext);

 RestServerComponent lRestServerComponent = new RestServerComponent(lSessionFactory);
 lRestServerComponent.getClients().add(Protocol.HTTPS​);

 lServletAdapter.setNext(lRestServerComponent);
 lServletAdapter.service(pRequest, pResponse);'
```

This solves the component issue (i think...), but now there's missing support for client Protocol.HTTPS. The post at stackoverflow states that i have to add additional jar files;
* org.restlet.ext.net org.restlet.ext.httpclient
* org.restlet.ext.ssl

Is this correct? How do i include this jar files with maven? 
Do i have to use the J2EE package instead of GAE?
```java
    <dependency>
            <groupId>org.restlet.jee</groupId>
            <artifactId>org.restlet.ext.httpclient</artifactId>
            <version>2.3.1</version>
        </dependency>

        <dependency>
            <groupId>org.restlet.jee</groupId>
            <artifactId>org.restlet.ext.ssl</artifactId>
            <version>2.1.7</version>
        </dependency>
```

Regards,

Roland Beuker

**Comment by jlouvel** [comment]
Hello, did you add the parameter to the Servlet config?

                 <init-param>
                         <param-name>org.restlet.clients</param-name>
                         <param-value>HTTP HTTPS FILE</param-value>
                 </init-param>

**Comment by rcbeuker** [comment]
Hello Jerome,

I'm working with Guice and therefore i don't have a Servlet config in my web.xml

```
<?xml version="1.0" encoding="UTF-8"?>
<web-app xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://java.sun.com/xml/ns/javaee"
         xsi:schemaLocation="http://java.sun.com/xml/ns/javaee http://java.sun.com/xml/ns/javaee/web-app_2_5.xsd"
         version="2.5">

    <display-name>Project</display-name>

    <filter>
        <filter-name>guiceFilter</filter-name>
        <filter-class>com.google.inject.servlet.GuiceFilter</filter-class>
    </filter>

    <filter-mapping>
        <filter-name>guiceFilter</filter-name>
        <url-pattern>*</url-pattern>
    </filter-mapping>

    <listener>
        <listener-class>com.ludus.server.guice.MyGuiceServletContextListener</listener-class>
    </listener>

    <welcome-file-list>
        <welcome-file>ludus.html</welcome-file>
    </welcome-file-list>

</web-app>
```
And the connected DispatchServletModule
```
public class DispatchServletModule extends ServletModule
{
    @Override
    public void configureServlets()
    {
        serve("/rest/*").with(RestletServlet.class);
    }
}
```
How do i configure the extra parameters with Guice?
Do i need this extra paramaters? Because i already added the HTTPS protocol in my servlet;
```
 RestServerComponent lRestServerComponent = new RestServerComponent(lSessionFactory);
 lRestServerComponent.getClients().add(Protocol.HTTPS​);
```
What about the missing/additional jar files ?;

* org.restlet.ext.net org.restlet.ext.httpclient
* org.restlet.ext.ssl

Regards,

Roland Beuker

**Comment by rcbeuker** [comment]
I checked Class ServerServlet and indeed ```"<param-value>HTTP HTTPS</param-value>"``` just adds the protocol;
```
 // Define the list of supported client protocols.
            final String clientProtocolsString = getInitParameter(CLIENTS_KEY,
                    null);
            if (clientProtocolsString != null) {
                final String[] clientProtocols = clientProtocolsString
                        .split(" ");
                Client client;

                for (final String clientProtocol : clientProtocols) {
                    client = new Client(clientProtocol);

                    if (client.isAvailable()) {
                        component.getClients().add(client);
                    } else {
                        log("[Restlet] Couldn't find a client connector for protocol "
                                + clientProtocol);
                    }
                }
            }
```

But my " if (client.isAvailable()) " returns false because there is no helper provider by the implementation.
Which libraries need to be added? Shouldn't these libraries be part of the GAE package?

Regards,

Roland

**Comment by rcbeuker** [comment]
I fixed it (actually Koen Maes did). The problem was the use of ``` "org.restlet.ext.net"```
```
<dependency>
  <groupId>org.restlet.gae</groupId>
  <artifactId>org.restlet.ext.net</artifactId>
  <version>2.2-M6</version>
 </dependency>
```
I added this because the book Restlet in Action specifies in paragraph 8.2.5

>Because GAE makes it possible to use its URL Fetch service through the synchronous
HttpURLConnection class, you’ll use the org.restlet.ext.net extension when
accessing external URLs by putting that extension’s JAR file in the classpath of the GAE
application. Client requests are then automatically executed through this connector.

Just removed this package and everything works :)
